packages/react/src/Spinner/Spinner.tsx

```diff
@@ -87,17 +87,21 @@ const StyledComponentSpinner = styled(Spinner)`
   ${sx}
 `
 
-function StyledSpinner({sx, ...props}: SpinnerProps) {
+const StyledBaseSpinner = styled.div`
+  ${sx}
+`
+
+function StyledSpinner({sx, className, ...props}: SpinnerProps) {
   const enabled = useFeatureFlag('primer_react_css_modules_team')
   if (enabled) {
     if (sx) {
-      return <Box sx={sx} as={Spinner} className={classes.SpinnerAnimation} {...props} />
+      return <StyledBaseSpinner sx={sx} as={Spinner} className={clsx(className, classes.SpinnerAnimation)} {...props} />
     }
 
-    return <Spinner className={classes.SpinnerAnimation} {...props} />
+    return <Spinner className={clsx(className, classes.SpinnerAnimation)} {...props} />
```

Comment:
Not related to the bug, but was pointed out that className was overwriting the CSS modules classname when being passed in, so I added this